integration/users/connection_test.go

```diff
@@ -307,6 +308,181 @@ func TestAuthenticationOnAuthenticatedConnection(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestAuthenticationAuthSource(t *testing.T) {
+	t.Parallel()
+
+	s := setup.SetupWithOpts(t, nil)
+	ctx, db := s.Ctx, s.Collection.Database()
+
+	for name, tc := range map[string]struct {
+		username         string
+		password         string
+		authenticationDB string
+		path             string
```

Comment:
The path component of MongoDB URI is officially called "defaultauthdb". Having those two separate fields makes it more complicated.

I wonder if having only two fields (a full URI with ":%d" for port and "current database") will make things easier.

Reply:
The test case is easier to read, body is slightly more complicated but I think it's better